src/OmniSharp.Abstractions/Models/v2/BlockSpan.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace OmniSharp.Models.V2
+{
+    public class BlockSpan
+    {
+        public BlockSpan(Range textSpan, Range hintSpan, string bannerText,  string type)
+        {
+            TextSpan = textSpan;
+            HintSpan = hintSpan;
+            BannerText = bannerText;
+            Type = type;
+        }
+
+        /// <summary>
+        /// The span of text to collapse.
+        /// </summary>
+        public Range TextSpan { get; }
+
+        /// <summary>
+        /// The span of text to display in the hint on mouse hover.
+        /// </summary>
+        public Range HintSpan { get; }
+
+        /// <summary>
+        /// The text to display inside the collapsed region.
+        /// </summary>
+        public string BannerText { get; }
+
+        public string Type { get; }
```

Comment:
What is this? What values are legal?

Reply:
This is property on the internal Roslyn types that's also stringly-typed. I wasn't sure if we would need to use it at some point so I left it in, but I can also remove it.

Reply:
For example, we should provide the possible values here so that OmniSharp consumers understand what it's used for.